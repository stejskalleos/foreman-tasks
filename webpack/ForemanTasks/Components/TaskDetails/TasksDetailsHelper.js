// Get Task ID from URL. Split url by '/', filter non-empty values and get Task ID.
export const getTaskID = () =>
  window.location.pathname
    .split('/')
    .filter(i => i)
    .slice(-1)[0];
