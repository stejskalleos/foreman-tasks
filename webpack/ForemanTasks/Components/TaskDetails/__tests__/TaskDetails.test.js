import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import TaskDetails from '../TaskDetails';

const fixtures = {
  'render without Props': {},
};

delete window.location;
window.location = new URL(
  'https://foreman.com/foreman_tasks/tasks/a15dd820-32f1-4ced-9ab7-c0fab8234c47/'
);

describe('TaskDetails', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TaskDetails, fixtures));
});
