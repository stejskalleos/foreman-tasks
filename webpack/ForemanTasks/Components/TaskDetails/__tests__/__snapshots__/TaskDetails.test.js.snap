// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetails rendering render without Props 1`] = `
<div
  className="task-details-react well"
>
  <ForwardRef
    animation={false}
    defaultActiveKey={1}
    id="task-details-tabs"
  >
    <Tab
      eventKey={1}
      title="Task"
    >
      <Task
        action=""
        allowDangerousActions={false}
        cancelTaskRequest={[Function]}
        cancellable={false}
        dynflowEnableConsole={false}
        endedAt=""
        error={Array []}
        executionPlan={Object {}}
        externalId=""
        failedSteps={Array []}
        fetchTaskDetails={[Function]}
        hasSubTasks={false}
        help=""
        id="a15dd820-32f1-4ced-9ab7-c0fab8234c47"
        input={Array []}
        label=""
        locks={Array []}
        output={Object {}}
        parentTask=""
        progress={0}
        refetchTaskDetails={[Function]}
        result="error"
        resumable={false}
        resumeTaskRequest={[Function]}
        runningSteps={Array []}
        showForceUnlockModal={false}
        showUnlockModal={false}
        startAt=""
        startBefore=""
        startedAt=""
        state=""
        taskReload={false}
        taskReloadStart={[Function]}
        taskReloadStop={[Function]}
        timeoutId={null}
        toggleForceUnlockModal={[Function]}
        toggleUnlockModal={[Function]}
        username=""
        usernamePath=""
      />
    </Tab>
    <Tab
      eventKey={2}
      title="Running Steps"
    >
      <RunningSteps
        executionPlan={Object {}}
        runningSteps={Array []}
      />
    </Tab>
    <Tab
      eventKey={3}
      title="Errors"
    >
      <Errors
        executionPlan={Object {}}
        failedSteps={Array []}
      />
    </Tab>
    <Tab
      eventKey={4}
      title="Locks"
    >
      <Locks
        locks={Array []}
      />
    </Tab>
    <Tab
      eventKey={5}
      title="Raw"
    >
      <Raw
        endedAt=""
        externalId=""
        id="a15dd820-32f1-4ced-9ab7-c0fab8234c47"
        input={Array []}
        label=""
        output={Object {}}
        startedAt=""
      />
    </Tab>
  </ForwardRef>
</div>
`;
